src/service-providers/mailchimp/ProviderSidebar.js

```diff
@@ -2,13 +2,81 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Fragment, useEffect } from '@wordpress/element';
-import { ExternalLink, SelectControl, Spinner, Notice } from '@wordpress/components';
+import { Fragment, useEffect, useState } from '@wordpress/element';
+import {
+	ExternalLink,
+	SelectControl,
+	Spinner,
+	Notice,
+	FormTokenField,
+} from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
-import { getListInterestsSettings } from './utils';
+import { getListInterestsSettings, getListTags, getTagNames, getTagIds } from './utils';
+
+const SegmentsSelection = ( {
+	onUpdate,
+	inFlight,
+	chosenInterestId,
+	availableInterests,
+	chosenTags,
+	availableTags,
+} ) => {
+	const [ segmentsData, setSegmentsData ] = useState( {
+		interest_id: chosenInterestId,
+		tag_ids: chosenTags,
+	} );
+	const updateSegmentsData = data => setSegmentsData( { ...segmentsData, ...data } );
+
+	// Update with real data after local (optimistic) update.
+	useEffect(() => {
+		setSegmentsData( {
+			interest_id: chosenInterestId,
+			tag_ids: chosenTags,
+		} );
+	}, [ chosenInterestId, chosenTags.join() ]);
+
+	const [ isInitial, setIsInitial ] = useState( true );
+	useEffect(() => {
+		if ( ! isInitial ) {
+			onUpdate( segmentsData );
+		}
+		setIsInitial( false );
+	}, [ segmentsData.interest_id, segmentsData.tag_ids.join() ]);
+
+	return (
+		<Fragment>
+			{ availableInterests.length && (
+				<SelectControl
+					label={ __( 'Groups', 'newspack-newsletters' ) }
+					value={ segmentsData.interest_id }
+					options={ [
+						{
+							label: __( '-- Select a group --', 'newspack-newsletters' ),
+							value: 'no_interests',
+						},
+						...availableInterests,
+					] }
+					onChange={ interest_id => updateSegmentsData( { interest_id } ) }
+					disabled={ inFlight }
+				/>
+			) }
+			{ availableTags.length && (
+				<FormTokenField
```

Comment:
The help text beneath the `FormTokenField` reads "Separate with commas or the Enter key," which doesn't seem to apply to this use case? It looks like it can't be changed or removed idiomatically (https://github.com/WordPress/gutenberg/blob/master/packages/components/src/form-token-field/index.js#L660-L669) so we might need to hide with CSS for now.

Reply:
Fixed in a4c86988655e74e99319feb723fa23acfe5e0a81